sdk/nodejs/runtime/resource.ts

```diff
@@ -582,12 +632,20 @@ async function prepareResource(label: string, res: Resource, parent: Resource |
             propertyToDirectDependencyURNs.set(propertyName, urns);
         }
 
+        const monitorSupportsStructuredAliases = await monitorSupportsAliasSpecs();
+        let computedAliases;
+        if (!monitorSupportsStructuredAliases && parent) {
+            computedAliases = allAliases(opts.aliases || [], name!, type!, parent, parent.__name!);
+        } else {
+            computedAliases = opts.aliases || [];
+        }
+
         // Wait for all aliases. Note that we use `res.__aliases` instead of `opts.aliases` as the former has been processed
         // in the Resource constructor prior to calling `registerResource` - both adding new inherited aliases and
         // simplifying aliases down to URNs.
         const aliases = [];
-        const uniqueAliases = new Set<string>();
-        for (const alias of (res.__aliases || [])) {
+        const uniqueAliases = new Set<Alias | URN>();
```

Comment:
It's a little strange having `Set<Alias>` given that `Alias` is an object type which I believe means object reference equality will be used to compare what's already in the `Set<Alias>`.

I wonder if rather than keeping this code here, it should move into the `!monitorSupportsStructuredAliases && parent` case above, e.g. something like (I'm writing in GH without compiler help):

```typescript
let aliases;
if (!monitorSupportsStructuredAliases && parent) {
    const computedAliases = allAliases(opts.aliases || [], name!, type!, parent, parent.__name!);
    aliases = [];
    const uniqueAliases = new Set<URN>();
    for (const alias of computedAliases) {
        const aliasVal = await output(alias).promise();
        if (!uniqueAliases.has(aliasVal)) {
            uniqueAliases.add(aliasVal);
            aliases.push(aliasVal);
        }
    }
} else {
    aliases = opts.aliases || [];
}
```

Reply:
Ah, missed that you'd still need to loop through and do `const aliasVal = await output(alias).promise();` for the non-computed case. I think it's simpler to leave it the way you had it.